x-pack/plugins/lens/public/types.ts

```diff
@@ -132,17 +132,17 @@ export interface VisualizationProps<T = unknown> {
 
 export interface SuggestionRequest<T = unknown> {
   // Roles currently being used
-  roles: DimensionRole[];
+  roles?: DimensionRole[];
```

Comment:
This confused me at first. I think we should come up with a different name, because "role"  is generally used in the context of security. It's not *wrong*, just misleading to me.

Also, I don't really understand how this is intended to work. It seems like roles would need to be associated with table columns in some way, and with this implementation, they are disconnected.

Reply:
As discussed offline, roles are not optimal for transferring meta information from one visualization to the next one.

I would remove them for now and once we actually hit this problem, we can think of a more suitable solution. ping @wylieconlon

Reply:
Agreed